test/unit/scripts/upgrade/TimeConstraints.t.sol

```diff
@@ -74,57 +76,108 @@ contract TimeConstraintsTest is Test {
     }
 
-    function test_checkExecuteWithinDayTime_InvalidRange() public {
+    function test_checkTimeWithinDayTime_InvalidRange() public {
         vm.expectRevert(
             abi.encodeWithSelector(
                 TimeConstraints.InvalidDayTimeRange.selector, Durations.from(18 hours), Durations.from(12 hours)
             )
         );
 
-        timeConstraints.checkExecuteWithinDayTime(Durations.from(18 hours), Durations.from(12 hours));
+        timeConstraints.checkTimeWithinDayTime(Durations.from(18 hours), Durations.from(12 hours));
+    }
+
+    // checkTimeWithinDayTimeAndEmit
+
+    function test_checkTimeWithinDayTimeAndEmit_HappyPath() external {
+        vm.warp(BASE_TIMESTAMP + 15 hours);
+        vm.expectEmit(true, true, true, true);
+        emit TimeConstraints.PerformedWithinDayTime(Durations.from(12 hours), Durations.from(18 hours));
+        timeConstraints.checkTimeWithinDayTimeAndEmit(Durations.from(12 hours), Durations.from(18 hours));
+
+        vm.warp(BASE_TIMESTAMP + 12 hours);
+        vm.expectEmit(true, true, true, true);
+        emit TimeConstraints.PerformedWithinDayTime(Durations.from(12 hours), Durations.from(18 hours));
+        timeConstraints.checkTimeWithinDayTimeAndEmit(Durations.from(12 hours), Durations.from(18 hours));
+
+        vm.warp(BASE_TIMESTAMP + 18 hours);
+        vm.expectEmit(true, true, true, true);
+        emit TimeConstraints.PerformedWithinDayTime(Durations.from(12 hours), Durations.from(18 hours));
+        timeConstraints.checkTimeWithinDayTimeAndEmit(Durations.from(12 hours), Durations.from(18 hours));
+
+        vm.warp(BASE_TIMESTAMP + 12 hours);
+        vm.expectEmit(true, true, true, true);
+        emit TimeConstraints.PerformedWithinDayTime(Durations.from(12 hours), Durations.from(12 hours));
+        timeConstraints.checkTimeWithinDayTimeAndEmit(Durations.from(12 hours), Durations.from(12 hours));
     }
 
-    function testFuzz_checkExecuteAfterTimestamp_HappyPath(uint32 secondsAfter) public {
+    // checkTimeAfterTimestamp
+
+    function testFuzz_checkTimeAfterTimestamp_HappyPath(uint32 secondsAfter) public {
         vm.warp(BASE_TIMESTAMP + secondsAfter);
-        timeConstraints.checkExecuteAfterTimestamp(Timestamps.from(BASE_TIMESTAMP));
+        timeConstraints.checkTimeAfterTimestamp(Timestamps.from(BASE_TIMESTAMP));
     }
 
-    function testFuzz_checkExecuteAfterTimestamp_NotReached(uint32 secondsBefore) public {
+    function testFuzz_checkTimeAfterTimestamp_NotReached(uint32 secondsBefore) public {
         vm.assume(secondsBefore < BASE_TIMESTAMP && secondsBefore > 0);
 
         vm.warp(BASE_TIMESTAMP - secondsBefore);
         vm.expectRevert(
             abi.encodeWithSelector(TimeConstraints.TimestampNotReached.selector, Timestamps.from(BASE_TIMESTAMP))
         );
-        timeConstraints.checkExecuteAfterTimestamp(Timestamps.from(BASE_TIMESTAMP));
+        timeConstraints.checkTimeAfterTimestamp(Timestamps.from(BASE_TIMESTAMP));
```

Comment:
`vm.expectRevert()` should be used with external call, to work correctly in the new forge versions.